docs/spec_draft.md

```diff
@@ -543,6 +544,124 @@ produces an implementation-defined value.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.dot_general
+
+### Semantics
+
+Performs general dot products between vectors, vector/matrix and matrix/matrix
+multiplication over dimensions specified in `dot_dimension_numbers`.
+
+### Operands
+
+| Name                    | Type                                                                                                                                                                                                 |
+|-------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `lhs`                   | tensor of integer, floating-point or complex element types                                                                                                                                           |
+| `rhs`                   | tensor of integer, floating-point or complex element types                                                                                                                                           |
+| `dot_dimension_numbers` | attribute containing the following 1-dimensional array of `si64`:<br>* `lhs_batching_dimensions`<br>* `rhs_batching_dimensions`<br>*  `lhs_contracting_dimensions`<br>* `rhs_contracting_dimensions` |
+| `precision_config`      | attribute containing the following precision(s):<br>* `STABLEHLO_PRECISION_DEFAULT`<br>* `STABLEHLO_PRECISION_HIGH`<br>* `STABLEHLO_PRECISION_HIGHEST`                                               |
+
+### Results
+
+| Name     | Type                                                       |
+|----------|------------------------------------------------------------|
+| `result` | tensor of integer, floating-point or complex element types |
+
+### Constraints
+
+  * (C1) `lhs`, `rhs` and `result` have the same element type.
```

Comment:
IMO, for StableHLO DotGeneralOp, we can consider the preferred element type to be same as the output element type.

According https://github.com/tensorflow/tensorflow/blob/9315da3fe49a8b36eadcc7e539b8692012320303/tensorflow/compiler/xla/service/shape_inference.cc#L768

should we write that the result type needs to be wider than any element type of the operand only for non floating-point type.

Reply:
I've updated the comment regarding some doubts I had, and it makes sense to use the result element type as the `preferred_element_type`, and thanks for catching this. XLA doc does not mention the use of `preferred_element_type` either. I'm wondering if this was left out deliberately since this goes into how the hardware computes it.